### PR TITLE
ref(delete org): Pull out delete code into own fn

### PR DESCRIPTION
**Pull request by ceorourke**
I want to be able to reuse this code in getsentry, so pull it out into another function. Lmk if there's a cleaner / better way to do this.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_details.py [hunk 1]
```diff
@@ -481,4 +464,28 @@ def delete(self, request, organization):
...
+        """
+        Delete an Organization
+        ``````````````````````
+
+        Schedules an organization for deletion.  This API endpoint cannot
+        be invoked without a user context for security reasons.  This means
+        that at present an organization can only be deleted from the
+        Sentry UI.
+
+        Deletion happens asynchronously and therefore is not immediate.
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_details.py]: Thanks for fixing the typo.

File: src/sentry/api/endpoints/organization_details.py [hunk 2]
```diff
@@ -481,4 +464,28 @@ def delete(self, request, organization):
...
+        However once deletion has begun the state of a project changes and
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_details.py]: Old problem, but why does this mention projects?

File: src/sentry/api/endpoints/organization_details.py [hunk 3]
```diff
@@ -414,25 +414,7 @@ def put(self, request, organization):
...
-
-        Deletion happens asynchronously and therefor is not immediate.
-        However once deletion has begun the state of a project changes and
-        will be hidden from most public views.
-
-        :pparam string organization_slug: the slug of the organization the
-                                          team should be created for.
-        :auth: required, user-context-needed
-        """
+    def handle_delete(self, request, organization):
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_details.py]: Would be good to explain that this method exists as a way for getsentry to override this endpoint with less duplication.

File: src/sentry/api/endpoints/organization_details.py (hunk 2 above)
Comment (ceorourke) [reply src/sentry/api/endpoints/organization_details.py]: I'm guessing whoever wrote this copy/pasta'd the docstring from here: https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/project_details.py#L644

**Comment by wedamija** [comment]
How will you actually be reusing this in getsentry?

**Comment by ceorourke** [comment]
@wedamija so I've been working on adding additional logging to stuff in _admin, one of those being closing an account (https://github.com/getsentry/getsentry/blob/master/static/getsentry/gsAdmin/views/customerDetails.jsx#L61)  which goes to this endpoint. I want to intercept the current path, log my stuff, and then call the delete code.

**Comment by wedamija** [comment]
> @wedamija so I've been working on adding additional logging to stuff in _admin, one of those being closing an account (https://github.com/getsentry/getsentry/blob/master/static/getsentry/gsAdmin/views/customerDetails.jsx#L61) which goes to this endpoint. I want to intercept the current path, log my stuff, and then call the delete code.

That makes sense, so you'll be doing something like inheriting from the Endpoint then calling this as part of your subclass?

**Review by wedamija — APPROVED** [review]

**Comment by ceorourke** [comment]
> > @wedamija so I've been working on adding additional logging to stuff in _admin, one of those being closing an account (https://github.com/getsentry/getsentry/blob/master/static/getsentry/gsAdmin/views/customerDetails.jsx#L61) which goes to this endpoint. I want to intercept the current path, log my stuff, and then call the delete code.
> 
> That makes sense, so you'll be doing something like inheriting from the Endpoint then calling this as part of your subclass?

yeah that's the idea :)

**Review by markstory — APPROVED** [review]

**Pull request merged**